functests/ptp/ptp.go

```diff
@@ -90,13 +93,20 @@ var _ = Describe("ptp", func() {
 	})
 
 	var _ = Describe("Test Offset", func() {
+		slaveLabel := ptpSlaveNodeLabel
 		BeforeEach(func() {
-			nodes, err := client.Client.Nodes().List(metav1.ListOptions{
-				LabelSelector: ptpSlaveNodeLabel,
-			})
-			Expect(err).ToNot(HaveOccurred())
-			if len(nodes.Items) < 1 {
-				Skip(fmt.Sprintf("PTP Nodes with label %s are not deployed on cluster", ptpSlaveNodeLabel))
+			if !execute.DiscoveryModeEnabled() {
+				nodes, err := client.Client.Nodes().List(metav1.ListOptions{
+					LabelSelector: ptpSlaveNodeLabel,
+				})
+				Expect(err).ToNot(HaveOccurred())
+				if len(nodes.Items) < 1 {
+					Skip(fmt.Sprintf("PTP Nodes with label %s are not deployed on cluster", ptpSlaveNodeLabel))
+				}
+			}
+			slaveLabel = checkPolicies(ptpLinuxDaemonNamespace)
+			if slaveLabel == "" {
+				Skip("No node with PTP labels found in discovery mode")
```

Comment:
This could be "No nodes configured as ptp slaves found on the cluster" or something along that line.
The more info we provide on skipping, the better it is since these tests will end up also in users' hands.

Reply:
Done